### PR TITLE
The function batch normalization seem not implemented in first hand  in ivy. 

### PR DESCRIPTION
**Pull request by chanpreet007**
Batch Normalisation Missing

### CODE REVIEW

**Comment by chanpreet007** [comment]
@jkeane508 
I think this should work.
def batch_norm(x, axis=-1, eps=1e-5, momentum=0.1, affine=True, training=True):
    if affine:
        x, gamma, beta = ivy.normalizations.weight_norm(x, axis=axis)
    else:
        gamma, beta = None, None
    moving_mean = ivy.variable(ivy.zeros(x.shape[axis]), requires_grad=False)
    moving_var = ivy.variable(ivy.ones(x.shape[axis]), requires_grad=False)
    if training:
        mean = ivy.mean(x, axis=axis, keepdims=True)
        var = ivy.variance(x, axis=axis, keepdims=True)
        x = (x - mean) / ivy.sqrt(var + eps)
        moving_mean = (1 - momentum) * moving_mean + momentum * ivy.squeeze(mean)
        moving_var = (1 - momentum) * moving_var + momentum * ivy.squeeze(var)
    else:
        x = (x - moving_mean) / ivy.sqrt(moving_var + eps)
    if gamma is not None and beta is not None:
        x = gamma * x + beta
    return x

**Comment by jkeane508** [comment]
hey @chanpreet007,

It looks like the force push has messed this PR up a bit, would you mind creating a new one after updating your branch?

Thanks,
James
